app/Helpers/shortcode.php

```diff
@@ -113,6 +113,13 @@ function normalize_user_shortcodes(string $value): string
 
 function normalize_game_shortcodes(string $value): string
 {
+    // If the original string contains a subpath beyond the game ID,
+    // eg: "https://retroachievements.org/game/5236/hashes", return the
+    // original string rather than attempting to normalize it.
+    if (preg_match("~https?://[\w\-\.]+/game/\d+/.+~si", $value)) {
+        return $value;
+    }
```

Comment:
Why isn't this also a problem with users?
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/9ccee5a7-88b3-4cf5-bad3-1dd35e3179fc)

And should we protect against similar behavior for achievements and tickets in case they get trailing paths at some point in the future?

Reply:
Suggest adding a unit test for this logic.

Reply:
Users already had a lookahead in the regex, which led me to implement a much more elegant solution in latest.

I've trimmed down three of those normalization functions into a single reusable function. I've left some TODOs around the ticket normalization function, which may be useful for the work happening in https://github.com/RetroAchievements/RAWeb/pull/2375.

Additionally, I've added several tests to cover the behavior.